CMakeLists.txt

```diff
@@ -115,15 +123,43 @@ int main() {
   return 0;
 }" SNAPPY_HAVE_SSSE3)
 
-check_cxx_source_compiles("
+  check_cxx_source_compiles("
 #include <immintrin.h>
 int main() {
   return _bzhi_u32(0, 1);
 }" SNAPPY_HAVE_BMI2)
 
-include(CheckSymbolExists)
-check_symbol_exists("mmap" "sys/mman.h" HAVE_FUNC_MMAP)
-check_symbol_exists("sysconf" "unistd.h" HAVE_FUNC_SYSCONF)
+elseif (CMAKE_SYSTEM_PROCESSOR MATCHES "PPC64*|ppc64*|powerpc64*")
+
+# Ensure -mcpu=native , many distros still default to power8
+
+  set(CMAKE_REQUIRED_FLAGS "${CMAKE_REQUIRED_FLAGS} -mcpu=native")
```

Comment:
I'd only add mcpu=native is another arch specified compile flag isn't used - like https://github.com/facebook/rocksdb/pull/6713/

Reply:
This one is a bit more complex,  most of the distros currently are compiled for power8 so the default build flags are for -mcpu=power8.   While we could dynamically check via elf aux inputs or reading special regs if we had privs to do so,  the code shown on the facebook page is still all done at compile time and doesn't seem to have any benefit compared to using -mcpu=native so I'm not sure what the argument is for.   
 
It's a tradeoff whether you want the code added to check dynamically and add a branch to change codepath at runtime,    keep -mcpu=native or another method to check at compile time what cpu it's being built on,  or do nothing and assume the default flags are correct or the user has modified the CLAGS or CMAKE_C_FLAGS.

Reply:
Agree its a few tough tradeoffs here and its up to what the maintainers accept. My main concern if a distro ends up with a p9 build machine but still supports P8. `-mcpu=native` ends up after `-mcpu=power8` and even I'm not sure what happens then.

Alternatives, [__builtin_cpu_supports](https://gcc.gnu.org/onlinedocs/gcc/Basic-PowerPC-Built-in-Functions-Available-on-all-Configurations.html#index-_005f_005fbuiltin_005fcpu_005fsupports) or [\_\_target\_\_](https://gcc.gnu.org/onlinedocs/gcc/Common-Function-Attributes.html) attributes on the function.